### PR TITLE
Added option to use adapter with instance profiles or credentials con…

### PR DESCRIPTION
**Pull request by maytanthegeek**
…figured in VM.

### CODE REVIEW

**Comment by AmitPhulera** [comment]
Hey @maytanthegeek,

Good job in implementing the support for using Roles. 😄 

Can we make it more generic using [AWS.CredentialsProviderChain](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/CredentialProviderChain.html)?

That will remove dependency of adding an extra parameter in the configuration. What are your views on that?

**Comment by maytanthegeek** [comment]
Have added support for [AWS.CredentialsProviderChain](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/CredentialProviderChain.html).

Now if credentials are passed in arguments then they will be added in the provider chain and will take precedence over other methods. If not passed, the general provider chain priority is used.

So, in case you want to use roles or already configured credentials in your environment, then you need not pass any keys explicitly.

File: index.js [hunk 1]
```diff
@@ -383,9 +395,9 @@ module.exports = {
     if (_.isUndefined(dsEntry)) {
       return done(
         new Error(
-          `Consistency violation: Cannot do that with datastore (${datastoreName})
```
Comment (AmitPhulera) [inline index.js]: Hey,
Can we remove this whitespace?

**Review by AmitPhulera — CHANGES_REQUESTED** [review]
Hey!
Overall the changes look good, there are some extra whitespaces which are just polluting the review, can you please look after them?

File: index.js (hunk 1 above)
Comment (maytanthegeek) [reply index.js]: Already done

Comment (maytanthegeek) [reply index.js]: Corrected those format changes. Had to do a force push.

**Review by AmitPhulera — APPROVED** [review]
Looks good.

**Pull request merged**